src/agni/server/receiver/InfoRequestReceiver.java

```diff
@@ -1,17 +1,41 @@
 package agni.server.receiver;
 
+import java.nio.ByteBuffer;
+import java.util.Vector;
+
 public class InfoRequestReceiver implements MessageParser {
-    private InfoListener[] infoListeners;
+    private Vector <InfoListener> infoListeners = null;
 
     public InfoRequestReceiver() {
-
+        infoListeners = new Vector<InfoListener>();
     }
 
-    private void notifyInfoRequest() {
+    private void notifyInfoRequest(String ip, byte type) {
```

Comment:
Since the type is parsed and immediately passed I don't think it makes sense to translate it to an enum then immediately translate it back to bytes for passing. I think it will just complicate the code.